src/javascript/app/base/header.js

```diff
@@ -350,30 +394,40 @@ const Header = (() => {
             }
         });
 
+        // Dynamic link for trader's hub cta for mobile menu
+        const mobile_platform_appstore_link     = getElementById('url-appstore');
+        const wallets_hub_link                  = Url.urlForDeriv('wallets', `ext_platform_url=${encodeURIComponent(window.location.href)}`);
+        const traders_hub_link                  = Url.urlForDeriv('appstore/traders-hub', `ext_platform_url=${encodeURIComponent(window.location.href)}`);
+        mobile_platform_appstore_link.href      = Client.hasWalletsAccount() ? wallets_hub_link : traders_hub_link;
+
         // Account Switcher Event
-        const acc_switcher                = getElementById('acc_switcher');
-        const account_switcher            = getElementById('account__switcher');
-        const account_switcher_dropdown   = getElementById('account__switcher-dropdown');
+        const acc_switcher                = Client.hasWalletsAccount() ? getElementById('wallet_switcher') : getElementById('acc_switcher');
+        const account_switcher            = Client.hasWalletsAccount() ? getElementById('wallet__switcher') : getElementById('account__switcher');
+        const account_switcher_dropdown   = Client.hasWalletsAccount() ? getElementById('wallet__switcher-dropdown') : getElementById('account__switcher-dropdown');
+        const account_switcher_active     = Client.hasWalletsAccount() ? 'wallet__switcher-dropdown--show' : 'account__switcher-dropdown--show';
+        const wallet_switcher_close       = getElementById('wallet__switcher-close');
```

Comment:
Does it (`wallet_switcher_close`) always exist or only if `Client.hasWalletsAccount() === true`?

Reply:
Because after you put an eventListener on it

Reply:
`wallet_switcher_close` exist only if Client.hasWalletaccount() is true. I remove the whole <WalletSwitcher /> from the DOM at the start in `switchHeader()` at the top.

Theres actually a lot of shared classnames/IDs in both `AccountSwitcher` and `WalletSwitcher`. Removing the switcher from the DOM was the way I could attach event handler without rewriting the same thing but with different classnames. JS didn't know where to attach the event handler when classnames/IDs are same.